design/design_template.md

```diff
@@ -0,0 +1,101 @@
+---
+title: neat-enhancement-idea
+target-version: release-X.X
+---
+
+# Neat Enhancement Idea
+
+This is the title of the enhancement. Keep it simple and descriptive. A good
+title can help communicate what the enhancement is and should be considered as
+part of any review.
+
+The YAML `title` should be lowercased and spaces/punctuation should be
+replaced with `-`.
+
+To get started with this template:
+1. **Pick a domain.** Find the appropriate domain to discuss your enhancement.
+1. **Make a copy of this template.** Copy this template into the directory for
```

Comment:
All of these steps are obvious, no? If they need steps on how to create a PR, we should do it in the development-flow.md.